src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/EventRoute.cs

```diff
@@ -552,6 +555,11 @@ internal void Clear()
         // Stores Source Items for separated trees
         private FrugalStructList<SourceItem> _sourceItemList;
 
+        // Stores arguments that are passed to TraceRoutedEvent.Trace (to reduce allocations)
+        private object[] _traceArguments;
+        private static readonly object _true = true;
```

Comment:
Can you use the `BooleanBoxes` here ?

Reply:
Great suggestion; didn't realise that class was available in WPF.